ql/src/java/org/apache/hadoop/hive/ql/HiveQueryLifeTimeHook.java

```diff
@@ -74,6 +75,15 @@ private void checkAndRollbackCTAS(QueryLifeTimeHookContext ctx) {
     PrivateHookContext pCtx = (PrivateHookContext) ctx.getHookContext();
     Path tblPath = pCtx.getContext().getLocation();
 
+    try {
+      FileSystem fs = FileSystem.get(conf);
```

Comment:
this will give you the defaultFs, the one specified by fs.defaultFs, when using federation or multiple FS, in that case it can lead to "Wrong FS" errors.
Do
```
tblPath.getFileSystem(conf)
```

Reply:
Done. Thanks for the review and info @ayushtkn .